services/ui_backend_service/cache/store.py

```diff
@@ -59,17 +61,21 @@ def __init__(self, event_emitter):
 
         # Bind an event handler for when we want to preload artifacts for
         # newly inserted content.
-        self.event_emitter.on("preload-artifacts", self.preload_event_handler)
-        self.event_emitter.on("run-parameters", self.run_parameters_event_handler)
+        if FEATURE_PREFETCH_ENABLE:
+            self.event_emitter.on("preload-artifacts", self.preload_event_handler)
+            self.event_emitter.on("run-parameters", self.run_parameters_event_handler)
```

Comment:
run-parameters event is produced by a table trigger. If this is skipped, then the timeline view in the UI never receives run parameters as they become available, instead requiring a page refresh.